src/lv_core/lv_obj.c

```diff
@@ -2250,7 +2250,7 @@ void * lv_obj_get_ext_attr(const lv_obj_t * obj)
  * @param obj pointer to an object which type should be get
  * @param buf pointer to an `lv_obj_type_t` buffer to store the types
  */
-void lv_obj_get_type(lv_obj_t * obj, lv_obj_type_t * buf)
+void lv_obj_get_type(const lv_obj_t * obj, lv_obj_type_t * buf)
```

Comment:
Why add `const` here if you cost back later?

Reply:
`obj->signal_cb` takes in a non-const `lv_obj_t *` since it may modify it. However, it (with near certainty, correct me if im wrong) will not modify it on `LV_SIGNAL_GET_TYPE` . So this is just to prevent a "discards const" error. I add `const` in the function signature to emphasize that the object isn't modified.

Reply:
I see and actually it's reasonable. Let's keep the API more descriptive.